src/App.js

```diff
@@ -1,23 +1,39 @@
-import logo from './logo.svg';
-import './App.css';
+import Footer from "./components/Footer";
+import "./colors.css";
+import "./global.css";
+import NavigationBar from "./components/NavigationBar";
+import Header from "./components/Header";
+import SharedPage from "./components/SharedPage";
+import SearchBar from "./components/SearchBar";
+import CardList from "./components/CardList";
+import ReadCard from "./components/ReadCard";
+import useGetFolder from "./utils/useGetFolder";
 
 function App() {
+  const { data } = useGetFolder();
+  const { profileImage, ownerName, folderName, links } = data || {};
+
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <NavigationBar />
+      <SharedPage
+        folderInfo={
+          <Header
+            profileImage={profileImage}
+            ownerName={ownerName}
+            folderInfo={folderName}
+          />
+        }
+        searchBar={<SearchBar />}
+        cardList={
+          <CardList>
+            {links?.map((link) => (
+              <ReadCard key={link?.id} {...link} />
+            ))}
+          </CardList>
+        }
+      />
```

Comment:
`SharedPage` 라는 컴포넌트를 더 적극 활용해보면 좋을 것 같아요.
지금은 App.js 에서 `useGetFolder ` 훅을 불러서 해당 훅의 반환값들을 그대로 SharedPage 에 넘겨주고 있는데,

그럴 필요 없이 SharedPage 컴포넌트 내부에서 직접 useGetFolder을 호출하면 되지 않을까요?
그러면 SharedPage에 어떤 컴포넌트가 들어가야하는지 ( Header, SearchBar, CardList) App.js 에서 알 필요 없을 것 같고요! 
App.js에서는 아래와 같은 형태로 SharedPage 라는 컴포넌트를 불러올 수 있도록 리팩토링 되면 좋을 것 같아요.

```suggestion
      <SharedPage />
```